madanalysis/dataset/dataset.py

```diff
@@ -1,335 +1,441 @@
+            "pdf_down_variation",
+            "scale_variation",
+            "pdf_variation",
+        ]:
             try:
                 tmp = float(value)
-            except:
-                logging.getLogger('MA5').error("the value of the attribute '"+variable+\
-                                               "' must be set to a positive floating-point number")
+            except ValueError:
+                logging.getLogger("MA5").error(
+                    "the value of the attribute '"
+                    + variable
+                    + "' must be set to a positive floating-point number"
+                )
                 return
-            if tmp>=0:
+            if tmp >= 0:
                 if variable == "scale_up_variation":
                     self.scaleup = tmp
                     if self.scaledn == None:
-                        self.scaledn = .0
+                        self.scaledn = 0.0
                 elif variable == "scale_down_variation":
                     self.scaledn = tmp
                     if self.scaleup == None:
-                        self.scaleup = 0.
+                        self.scaleup = 0.0
                 elif variable == "pdf_up_variation":
                     self.pdfup = tmp
                     if self.pdfdn == None:
-                        self.pdfdn = 0.
+                        self.pdfdn = 0.0
                 elif variable == "pdf_down_variation":
                     self.pdfdn = tmp
                     if self.pdfup == None:
-                        self.pdfup = 0.
+                        self.pdfup = 0.0
                 elif variable == "scale_variation":
                     self.scaledn = tmp
                     self.scaleup = tmp
                 elif variable == "pdf_variation":
                     self.pdfdn = tmp
                     self.pdfup = tmp
             else:
-                logging.getLogger('MA5').error("the value of the attribute '"+variable+\
-                                               "' must be set to a positive floating-point number")
+                logging.getLogger("MA5").error(
+                    "the value of the attribute '"
+                    + variable
+                    + "' must be set to a positive floating-point number"
+                )
+                return
+
+        # Dynamic scale choice
+        elif variable == "dynamic_scale_choice":
+            try:
+                tmp = int(value)
+                if tmp not in [1, 2, 3, 4]:
+                    raise ValueError("dynamic_scale_choice is not within 1,2,3,4")
+            except ValueError:
+                logging.getLogger("MA5").error(
+                    "Dynamic scale choice can only be 1,2,3 or 4."
+                )
                 return
+            self.dynamic_scale_choice = tmp
+
+        elif variable == "n_point_scale_variation":
+            try:
+                tmp = int(value)
+                if tmp not in [3, 5, 7, 9]:
```

Comment:
What is 5?

Reply:
couldn't remember how many there were so added one extra instead of one less